lib/ansible/executor/play_iterator.py

```diff
@@ -271,35 +292,36 @@ def _get_next_task_from_state(self, state, host):
                 return (state, None)
 
             if state.run_state == IteratingStates.SETUP:
-                # First, we check to see if we were pending setup. If not, this is
-                # the first trip through IteratingStates.SETUP, so we set the pending_setup
-                # flag and try to determine if we do in fact want to gather facts for
-                # the specified host.
-                if not state.pending_setup:
-                    state.pending_setup = True
+                # First, we check to see if we completed both setup tasks injected
```

Comment:
After staring at this for a while the `PlayIterator` changes look good to me.

Reply:
@mkrizek Thank you for the review!